src/framex/driver/ingress.py

```
@@ -146,3 +147,24 @@ async def inner(self) -> str:  # pragma: no cover
+        **kwargs: Any,
+    ) -> None:
+        method_set: set[str] = {m.upper() for m in (methods or [])}
+        norm_path = re.sub(r"\{[^}]+\}", "{}", path)
+
+        for route in app.routes:
+            if (
+                isinstance(route, APIRoute)
+                and re.sub(r"\{[^}]+\}", "{}", route.path) == norm_path
+                and route.methods & method_set
+            ):
+                raise RuntimeError(f"Duplicate API route: {sorted(method_set)} {norm_path}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)